.github/workflows/test_sh.yaml

```diff
@@ -0,0 +1,120 @@
+name: Run tests using test.sh
+
+on: [pull_request]
+
+jobs:
+  test:
+    name: Python ${{ matrix.os.python }} tests on ${{ matrix.os.name }}-${{ matrix.os.version }}
+    runs-on: ubuntu-latest
+
+    strategy:
+      fail-fast: false
+      matrix:
+        os:
+        - name: centos
+          version: 7
+          python: 2
+          engine: docker
+
+        - name: fedora
+          version: 30
+          python: 3
+          engine: docker
+
+        - name: fedora
+          version: 31
+          python: 3
+          engine: docker
+
+    steps:
+    - uses: actions/checkout@v1
+
+    - name: test.sh
+      env:
+        OS: ${{ matrix.os.name }}
+        OS_VERSION: ${{ matrix.os.version }}
+        PYTHON_VERSION: ${{ matrix.os.python }}
+        ENGINE: ${{ matrix.os.engine }}
+      run: ./test.sh
+
+    - name: Coveralls Python parallel
+      uses: AndreMiras/coveralls-python-action@v20200413
+      with:
+        parallel: true
```

Comment:
Does this not need the `GITHUB_TOKEN` secret?

Reply:
@chmeliik; `GITHUB_TOKEN` is the default

Reply:
Ohh, that makes sense :D